faulty_code.py

```
@@ -0,0 +1,86 @@
+    """Calculate discount. Bug: negative prices allowed, floating point issues."""
+    final = price - (price * discount / 100)
+    return round(final, 2)
+    return final  # unreachable code
+
+
+class UserSession:
+    """Session manager. Bug: mutable default arg, missing init for attributes."""
+
+    active_sessions = []
+
+    def __init__(self, user, permissions=[]):
```

Ruff (0.15.13)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)